copulas/multivariate/Tree.py

```diff
@@ -0,0 +1,379 @@
+            name, param = Copula.select_copula(self.u_matrix[:, 0],
+                                               self.u_matrix[:, ind])
+            new_edge = Edge(itr, 0, ind, tau_mat[0, ind], name, param)
+            new_edge.parent.append(0)
+            new_edge.parent.append(ind)
+            self.edge_set.append(new_edge)
+
+    def _build_kth_tree(self):
+        # find anchor variable with highest sum of dependence with the rest
+        temp = np.empty([self.n_nodes, 2])
+        temp[:, 0] = np.arange(self.n_nodes, dtype=int)
+        temp[:, 1] = np.sum(abs(self.tau_mat), 1)
+        tau_sorted = temp[temp[:, 1].argsort()[::-1]]
+        anchor = int(temp[0, 0])
+        self.tau_mat[anchor, :] = np.NaN
+        # sort the rest of variables based on dependence with anchor variable
+        aux = np.empty([self.n_nodes, 3])
+        aux[:, 0] = np.arange(self.n_nodes, dtype=int)
+        aux[:, 1] = self.tau_mat[:, anchor]
+        aux[:, 2] = abs(self.tau_mat[:, anchor])
+        aux[anchor, 2] = -10
+        aux_sorted = aux[aux[:, 2].argsort()[::-1]]
+        for itr in range(self.n_nodes - 1):
+            right = aux_sorted[itr, 0]
+            U1, U2 = self.prev_T.edge_set[anchor].U,
+            self.prev_T.edge_set[right].U
+            name, param = Copula.select_copula(U1, U2)
+            [ed1, ed2, ing] =\
+                self._identify_eds_ing(self.prev_T.edge_set[anchor],
+                                       self.prev_T.edge_set[right])
+            new_edge = Edge(itr, ed1, ed2, tau_sorted[itr, 1], name, param)
+            new_edge.parent.append(anchor)
+            new_edge.parent.append(right)
+            new_edge.parents.append(self.prev_T.edge_set[anchor])
+            new_edge.parents.append(self.prev_T.edge_set[right])
+            self.edge_set.append(new_edge)
+
+
+class DTree(Tree):
+    def __init__(self, k, n, tau_mat, prev_T):
+        super(DTree, self).__init__(k, n, tau_mat, prev_T)
+
+    def _build_first_tree(self):
+        # find the pair of maximum tau
+        tau_mat = self.tau_mat
+        np.fill_diagonal(tau_mat, np.NaN)
+        tau_y = tau_mat[:, 0]
+        temp = np.empty([self.n_nodes, 3])
+        temp[:, 0] = np.arange(self.n_nodes)
+        temp[:, 1] = tau_y
+        temp[:, 2] = abs(tau_y)
+        temp[np.isnan(temp)] = -10
+        tau_sorted = temp[temp[:, 2].argsort()[::-1]]
+        left_ind = tau_sorted[0, 0]
+        right_ind = tau_sorted[1, 0]
+        T1 = np.array([left_ind, 0, right_ind]).astype(int)
+        tau_T1 = tau_sorted[:2, 1]
+        # replace tau matrix of the selected variables as a negative number
+        tau_mat[:, [T1]] = -10
+        for k in range(2, self.n_nodes - 1):
+            valL, left = np.max(tau_mat[T1[0], :]),\
+                np.argmax(tau_mat[T1[0], :])
+            valR, right = np.max(tau_mat[T1[-1], :]),\
+                np.argmax(tau_mat[T1[-1], :])
+            if valL > valR:
+                # add nodes to the left'''
```

Comment:
The tripe quote simbol on the end of the line is unnecesary. Please delete it. Same with the comment on the block below